src/world/Script/NativeScriptApi.cpp

```diff
@@ -138,6 +137,11 @@ namespace Sapphire::ScriptAPI
   {
   }
 
+  Event::EventHandler::EventState EventScript::checkExtraConditions( Sapphire::Entity::Player& player, uint32_t eventId )
```

Comment:
probably want this to replicate the name of the struct a bit more, `checkExtraConditions` doesn't really mean anything obvious

Reply:
I think this should be a function that checks requirements that aren't in the exd data. So this should called after the generic availability check.
So are something like `checkSpecialRequirements` are a better name for this?
Function names aren't my strength xD

Reply:
i thought about this a bit and i would actually prefer that this is just called `getQuestAvailability` and the return is just like `QuestAvailability` - i can't think of any cases where this would apply to anything that isn't a quest in this case, and we can call different handlers for different cases as needed

i'd rather have more specific handlers that have a very specific scope rather than one general function that does a bunch of different things because that'll just be a nightmare over time

Reply:
Should this enum then stay in `EventHandler` or should I rather move it to `Common`?

Reply:
nah its fine where it is, it doesn't really make sense for it to be in common anyway given its only really used within the event system